src/dma/collector/workflows/collection_extractor/base.py

```diff
@@ -13,42 +13,96 @@
 # limitations under the License.
 from __future__ import annotations
 
+from datetime import datetime, timezone
 from typing import TYPE_CHECKING
 
 from rich.table import Table
+from sqlalchemy.ext.asyncio import AsyncSession
 
+from dma.__about__ import __version__ as current_version
+from dma.collector.dependencies import provide_collection_query_manager
 from dma.collector.workflows.base import BaseWorkflow
+from dma.lib.db.base import SourceInfo, get_engine
 from dma.lib.exceptions import ApplicationError
 
 if TYPE_CHECKING:
     from duckdb import DuckDBPyConnection
     from rich.console import Console
 
     from dma.collector.query_managers import CanonicalQueryManager, CollectionQueryManager
-    from dma.types import SupportedSources
 
 
 class CollectionExtractor(BaseWorkflow):
     def __init__(
         self,
         local_db: DuckDBPyConnection,
+        src_info: SourceInfo,
+        database: str,
         canonical_query_manager: CanonicalQueryManager,
-        collection_query_manager: CollectionQueryManager,
-        db_type: SupportedSources,
         console: Console,
+        collection_identifier: str | None,
     ) -> None:
-        self.collection_query_manager = collection_query_manager
-        super().__init__(local_db, canonical_query_manager, db_type, console)
+        self.src_info = src_info
+        self.database = database
+        self.collection_identifier = collection_identifier
+        super().__init__(local_db, canonical_query_manager, src_info.db_type, console)
 
     async def execute(self) -> None:
         await super().execute()
-        await self.extract_collection()
-        await self.process_collection()
+        execution_id = (
```

Comment:
another note; we should probably re-use the same execution ID for all databaes we collect from.  This will let us consolidate into a single set of files.  No need to do anything with this at the moment though since the "write to file" feature is not currently enabled.

Reply:
The current changes already reuse this execution ID for all the databases.